### PR TITLE
feat(astro): add eslint_d to pack.astro

### PR DESCRIPTION
**Pull request by rafaelderolez**
Use pack.typescript's setup to init eslint_d in pack.astro

### CODE REVIEW

File: lua/astrocommunity/pack/astro/init.lua [hunk 1]
```diff
@@ -17,4 +39,33 @@ return {
...
+        local null_ls = require "null-ls"
+        null_ls.register(
+          null_ls.builtins.diagnostics.eslint_d.with { condition = has_eslint, extra_filetypes = "astro" }
+        )
+        null_ls.register(
+          null_ls.builtins.formatting.eslint_d.with { condition = has_eslint, extra_filetypes = "astro" }
+        )
+        null_ls.register(
+          null_ls.builtins.code_actions.eslint_d.with { condition = has_eslint, extra_filetypes = "astro" }
+        )
```
Comment (Uzaaft) [inline lua/astrocommunity/pack/astro/init.lua:67]:
```suggestion
        local null_ls = require "null-ls"
        if null_ls.builtins.diagnostics.eslint_d then
        	utils.list_insert_unique(
	        null_ls.builtins.diagnostics.eslint_d.filetypes , "astro")
	else 
	null_ls.register(
          null_ls.builtins.diagnostics.eslint_d.with { condition = has_eslint, extra_filetypes = "astro" }
        )
        end
        if null_ls.builtins.code_actions.eslint_d then
        	utils.list_insert_unique(
        	null_ls.builtins.code_actions.eslint_d.filetypes , "astro")
        else
        	null_ls.register(
          null_ls.builtins.formatting.eslint_d.with { condition = has_eslint, extra_filetypes = "astro" }
        )
        if null_ls.builtins.formatting.eslint_d then 
        
        utils.list_insert_unique(
        null_ls.builtins.formatting.eslint_d.filetypes , "astro")
        else
        null_ls.register(
          null_ls.builtins.code_actions.eslint_d.with { condition = has_eslint, extra_filetypes = "astro" }
        )
        end
        
```

Comment (Uzaaft) [reply lua/astrocommunity/pack/astro/init.lua:67]: @rafaelderolez  I think this should work? 🤞🏾

Comment (rafaelderolez) [reply lua/astrocommunity/pack/astro/init.lua:67]: I'll commit your suggestion and give it a whirl 🫡

Comment (rafaelderolez) [reply lua/astrocommunity/pack/astro/init.lua]: @Uzaaft no that breaks eslint_d alltogether!

**Comment by rafaelderolez** [comment]
> @rafaelderolez I'm not getting any wiser on this one

Hey @Uzaaft. I haven't had the time to look into this further. I also moved away from using Astro.build on the side-project Iw as working. Maybe we should have an Issue with a `help-wanted` label if someone else wants to take a stab at this?

**Comment by Uzaaft** [comment]
@rafaelderolez  Let's go for creating an issue with help-wanted.

**Comment by Uzaaft** [comment]
@rafaelderolez Perhaps we could take inspiration from the prettier + md fix?

**Pull request closed without merging**